python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py

```diff
@@ -118,12 +118,26 @@ def __call__(
             set_status_on_exception=False,
         ) as span:
             agent = args[0] if args else None
+            if agent:
+                span.set_attribute(SpanAttributes.GRAPH_NODE_ID, agent.role)
+
             crew = agent.crew if agent else None
             task = instance
 
-            if crew:
+            if crew and agent:
                 span.set_attribute("crew_key", crew.key)
                 span.set_attribute("crew_id", str(crew.id))
+
+                # Find the current agent, the previous agent is the parent node
+                for i, a in enumerate(crew.agents):
```

Comment:
```suggestion
            if crew:
                span.set_attribute("crew_key", crew.key)
                span.set_attribute("crew_id", str(crew.id))

         if agent:
                # Find the current agent, the previous agent is the parent node
                for i, a in enumerate(crew.agents):
```
Isn't this what you need?

Reply:
I can re-order the code a little bit which I think will make this flow a little more obvious